[Ruppa Nagarajan Sivayoga Subramanian] iP

sivayogasubramanian opened the pull request:
## Jarvis
> “Productivity is never an accident. It is always the result of a commitment to excellence, intelligent planning, and focused effort.” – Paul J. Meyer ([source](https://dansilvestre.com/productivity-quotes/))

Jarvis frees your mind of having to remember things you need to do. It's,
* text-based
* easy to learn
* ~~FAST~~ _SUPER_ FAST to use

All you need to do is,

1. download it from [here](https://github.com/sivayogasubramanian/ip/releases/tag/v1.0).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE!**

Features:
- [x]  Managing tasks
- [x]  Managing deadlines
- [x]  CLI mode
- [x]  GUI mode
- [ ]  Reminders (coming soon)

If you **Java programmer**, you can use it to practice Java too. Here's the `Launcher`class:

```java
package jarvis;

import java.io.File;

import javafx.application.Application;

public class Launcher {
    /**
     * Main method for the Jarvis application.
     *
     * @param args Command line arguments.
     */
    public static void main(String ...args) {
        if (args.length > 0) {
            switch (args[0].toLowerCase()) {
            case "-c":
            case "--console":
                Launcher.launchCli();
                break;
            case "-g":
            case "--gui":
                Launcher.launchGui();
                break;
            default:
                System.out.println("Invalid CLI argument!");
            }
        }
        if (args.length == 0) {
            Launcher.launchGui();
        }
    }

    private static void launchCli() {
        Jarvis jarvis = new Jarvis("data" + File.separator + "jarvis.txt");
        jarvis.launchInCliMode();
    }

    private static void launchGui() {
        Application.launch(Main.class);
    }
}
```

Commit:
Add Gradle support

sivayogasubramanian committed:
Add Greet, Echo, Exit

sivayogasubramanian committed:
Customise chatbot name and personality

sivayogasubramanian committed:
Refactor package name

sivayogasubramanian committed:
Change access modifier to protected

sivayogasubramanian committed:
Implement add and list

sivayogasubramanian committed:
Add mark as done

sivayogasubramanian committed:
Add Todo, Deadline and Event subclasses

sivayogasubramanian committed:
Add action enums

sivayogasubramanian committed:
Add action type classes

sivayogasubramanian committed:
Implement adding of different task types

sivayogasubramanian committed:
Implement handling of errors

sivayogasubramanian committed:
Streamline string handling logic

sivayogasubramanian committed:
Change final declarations to non-final for non-constants

sivayogasubramanian committed:
Handle case when list is empty

sivayogasubramanian committed:
Improve list toString method

sivayogasubramanian committed:
Remove unused package imports

sivayogasubramanian committed:
Add automated text ui testing

sivayogasubramanian committed:
Implement deleting of tasks

sivayogasubramanian committed:
Fix typo in action object name

sivayogasubramanian committed:
Implement saving data to hard-disk

sivayogasubramanian committed:
Store deadline as LocaoDateTime object

sivayogasubramanian committed:
Merge branch 'branch-Level-7'

sivayogasubramanian committed:
Merge branch 'branch-Level-8'

sivayogasubramanian committed:
Fix errors after merging Level-8 branch

sivayogasubramanian committed:
Refactor according to tasks in A-MoreOOP

sivayogasubramanian committed:
Add JUint tests

sivayogasubramanian committed:
Add JavaDoc for all public methods

sivayogasubramanian committed:
Refactor code according to java coding standard

sivayogasubramanian committed:
Merge branch 'branch-A-Javadoc'

sivayogasubramanian committed:
Merge branch 'branch-A-CodingStandard'

sivayogasubramanian committed:
Implement Find

sivayogasubramanian committed:
Merge branch 'branch-Level-9'

sivayogasubramanian committed:
Refactor JavaDocs for Ui class

sivayogasubramanian committed:
Merge branch 'add-gradle-support' into branch-A-Gradle

sivayogasubramanian committed:
Add javafx dependencies and checkstyle

sivayogasubramanian committed:
Merge branch 'branch-A-Gradle'

sivayogasubramanian committed:
Change jar archive base name

sueann-chua commented on src/main/java/jarvis/command/Command.java:
```diff
@@ -0,0 +1,54 @@
+public abstract class Command {
+
+    /**
+     * Creates an appropriate command depending on the user input
```
Sentences in method headers should end with '.'

sueann-chua reviewed (commented):
Looks good! No coding standard violations other than missing fullstops at the end of sentences in method headers :)

sivayogasubramanian committed:
Add GUI support

sivayogasubramanian committed:
Fix unit test failing

sivayogasubramanian committed:
Customize greeting message

sivayogasubramanian committed:
Fix javadoc to follow coding standard

sivayogasubramanian committed:
Add javadoc for DialogBox

sivayogasubramanian committed:
Merge branch 'branch-Level-10'

sivayogasubramanian committed:
Use vararg in launcher main method

sivayogasubramanian committed:
Add newline to end of file

hanif-kamal reviewed (commented):
Really well done! Your code quality is super good and I really couldn't find anything that goes against the code quality guidelines.

Gordon25 commented on src/main/java/jarvis/Jarvis.java:
```diff
@@ -0,0 +1,88 @@
+        }
+    }
+
+    public String getResponse(String userInput) {
```
Perhaps add javadoc comment for this method here?

Gordon25 commented on src/main/java/jarvis/command/Command.java:
```diff
@@ -0,0 +1,56 @@
+    public static Command createCommand(String userInput) throws JarvisException {
+        CommandTypeEnum commandType = Parser.getCommandTypeFromInput(userInput);
+        String userInputWithoutCommandPrefix = Parser.getInputWithoutCommandType(userInput,
+                commandType.getCommandTypeStringLength());
```
I like the use of like break. 👍

Gordon25 commented on src/main/java/jarvis/task/Deadline.java:
```diff
@@ -0,0 +1,57 @@
+    }
+
+    /**
+     * String representation of a deadline task.
```
Would it be better if you phrased it in terms of gets ... .... or returns ... .... ?

Gordon25 commented on src/main/java/jarvis/task/TaskList.java:
```diff
@@ -0,0 +1,158 @@
+     *
+     * @return A string that is a summary of the task list
+     */
+    public String taskListSummary() {
```
Should this be renamed to start with an action instead? 🤔

Gordon25 commented on src/test/java/jarvis/message/ExitMessageTest.java:
```diff
@@ -0,0 +1,28 @@
+    private static final String DEFAULT_EXIT_MESSAGE = "Bye Bye! I am always ready when you need me!";
+
+    @Test
+    public void exitMessage_emptyConstructor_defaultExitMessage() {
```
I like the namings for the test methods. 👍

Gordon25 reviewed (commented):
Overall, besides some nits, LGTM! 😁

sivayogasubramanian committed:
Fix code quality issues

There was some code quality violations that was brought up during the PR
review. After looking through the codebase, it was found that other
similar issues were present.

Improve JavaDoc for methods and rename methods to better reflect their
action.

JavaDoc header statement should start with an action keyword according
to module standards. Methods also should be named after what it is doing
for better readability.

Refer to the PR request comments here:
https://github.com/nus-cs2103-AY2122S1/ip/pull/345

sivayogasubramanian committed:
Improve code quality according to module standards

sivayogasubramanian committed:
Add java assertions

sivayogasubramanian committed:
Use streams for toString and toStorageFormatString method

sivayogasubramanian committed:
Merge pull request #1 from sivayogasubramanian/branch-A-CodeQuality

Fix code quality issues

sivayogasubramanian committed:
Merge branch 'master' into branch-A-Assertions

sivayogasubramanian committed:
Merge pull request #2 from sivayogasubramanian/branch-A-Assertions

Add java assertions

sivayogasubramanian committed:
Merge branch 'master' into branch-A-Streams

sivayogasubramanian committed:
Merge pull request #3 from sivayogasubramanian/branch-A-Streams

Use streams for toString and toStorageFormatString method

TaskList toString and toStorageFormatString method uses for-loops to build the output string.

Java streams can be used here to make the code more elegant.

Let's
* use java streams and its related methods to build the string
* StringBuilder should still be used due to its better performance

sivayogasubramanian committed:
Fix checkstyle errors

sivayogasubramanian committed:
Add CI pipeline

sivayogasubramanian committed:
Merge pull request #4 from sivayogasubramanian/branch-A-CI

Set up Continuous Integration (CI)

This project does not have a CI pipeline.

There needs to be a pipeline to check the codebase before merging changes to the master branch.

Adding a gradle.yml file so that github actions can run.

sivayogasubramanian committed:
Implement undoing of commands

sivayogasubramanian committed:
Merge pull request #5 from sivayogasubramanian/branch-C-Undo

Implement undoing of commands

sivayogasubramanian committed:
Use datetime object for event

sivayogasubramanian committed:
Implement help command

sivayogasubramanian committed:
Fix checkstyle errors

sivayogasubramanian committed:
Fix code quality issues in javadocs

sivayogasubramanian committed:
Merge branch 'master' into finalise-features

sivayogasubramanian committed:
Merge pull request #7 from sivayogasubramanian/finalise-features

Finalise features for final submission

sivayogasubramanian committed:
Improve UI

sivayogasubramanian committed:
Add user guide

sivayogasubramanian committed:
Add Ui.png

sivayogasubramanian committed:
Set theme jekyll-theme-midnight

sivayogasubramanian committed:
Set theme jekyll-theme-midnight

sivayogasubramanian committed:
Set theme jekyll-theme-midnight

sivayogasubramanian committed:
Set theme jekyll-theme-cayman

sivayogasubramanian committed:
Improve code quality in Storage.java

sivayogasubramanian committed:
Merge branch 'master' of github.com:sivayogasubramanian/ip

sivayogasubramanian committed:
Fix minor bugs and improve help text

sivayogasubramanian committed:
Improve method header comments

sivayogasubramanian committed:
Fix bugs with Find and Undo commands

sivayogasubramanian committed:
Improve user guide

sivayogasubramanian committed:
Change images

sivayogasubramanian committed:
Use new Jarvis and User images

sivayogasubramanian committed:
Update docs to include image licence

sivayogasubramanian committed:
Merge pull request #8 from sivayogasubramanian/change-images

Change images and include licence in README.md

sivayogasubramanian committed:
Fix index out of bound error when exiting app
